src/main/java/core/basesyntax/TreeSetCharacters.java

```diff
@@ -17,7 +23,31 @@
  * Результат 2: acf</p>
  */
 public class TreeSetCharacters {
-    public String getUniqueCharacters(String fileName) {
-        return null;
+
+    private static final int CHARS_NUMBER = 5;
+
+    public String getUniqueCharacters(String fileName) throws IOException {
+        String resultString = "";
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fileName))) {
+            Set<Character> set = new TreeSet<>();
+            while (bufferedReader.ready()) {
+                String letters = bufferedReader.readLine().toLowerCase()
+                        .replaceAll("[^a-zA-Z]", "");
+                for (int i = 0; i < letters.length(); i++) {
+                    set.add(letters.charAt(i));
+                }
+            }
+            int size = set.size();
```

Comment:
we can simplify:
resultString = set.toString().replaceAll("[^a-zA-Z]", "");
            if (set.size() > CHARS_NUMBER) {
return resultString.substring(0, CHARS_NUMBER);
            }
            return resultString;

Reply:
Fixed.